qiskit_dynamics/backend/backend_utils.py

```diff
@@ -167,3 +168,92 @@ def _sample_probability_dict(
 def _get_counts_from_samples(samples: list) -> Dict:
     """Count items in list."""
     return dict(zip(*np.unique(samples, return_counts=True)))
+
+
+def _get_subsystem_probabilities(probability_tensor: np.ndarray, sub_idx: int) -> np.ndarray:
+    """Marginalize a probability vector according to subsystems.
```

Comment:
Can you put a note here that this is a temporary replacement for `QuantumState._subsystem_probabilities`. Or have you made changes now to make this even more efficient than if you used that method?

Also, make this doc string a bit more clear, something like: `Marginalize a probability array to a single subsystem.`.

Reply:
It's only a tiny bit more efficient than `QuantumState._subsystem_probabilities` but I guess the main reason for bringing it here is so we don't use a private static method.